cdci_data_analysis/analysis/instrument.py

```diff
@@ -128,24 +129,26 @@ def set_pars_from_dic(self, par_dic, verbose=False):
             query_obj = self.get_query_by_name(query_name)
             # loop over the list of parameters for the requested query,
             # but also of the instrument query and source query
-            for par in (query_obj.parameters +
-                        self.instrumet_query.parameters +
-                        self.src_query.parameters):
-                # this is required because in some cases a parameter is set without a name (eg UserCatalog),
-                # or they don't have to set (eg scw_list)
-                # 
-                if par.name is not None and par.name not in params_not_to_be_included:
-                    par.set_from_form(par_dic, verbose=verbose)
-
-                self.logger.info("set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s", par, par.name, par.value, par_dic.get(par.name, None))
-                if par.name == "scw_list":
-                    self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
-
+            param_list = (query_obj.parameters +
+                          self.instrumet_query.parameters +
+                          self.src_query.parameters)
         else:
-            for _query in self._queries_list:
-                for par in _query.parameters:
-                    if par.name is not None and par.name not in params_not_to_be_included:
-                        par.set_from_form(par_dic, verbose=verbose)
+            param_list = [par for _query in self._queries_list for par in _query.parameters]
+
+        for par in param_list:
+            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
+            # or they don't have to set (eg scw_list)
+            if par.name is not None and par.name not in params_not_to_be_included:
+                # set the value for par in to a default format,
+                # or to a default value if this is not included within the request
+                par.set_value_from_form(par_dic, verbose=verbose)
+            self.logger.info("set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
+                             par, par.name, par.value, par_dic.get(par.name, None))
+            if par.name == "scw_list":
+                self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
+        for par in param_list:
```

Comment:
why identical second loop?

Reply:
It's not identical, they serve two different purposes, that should be treated in separately:

- the first one takes care of setting the various values in the `form`, (for the parameters not present in the `form`, those are set to their default value), as well as in their default format (eg `isot` for `T1` and `T2`)
- the second sets the `units_name` in the `form` for those parameters where this is specified (eg `T1` and `T2`)

The reason for having two distinct loops is related to ho the assigning of the value to the `_astropy_time` ([here](https://github.com/oda-hub/dispatcher-app/blob/64e032b28fe80fbcf8ea2cda532e592befb6ecce/cdci_data_analysis/analysis/parameters.py#L487-L496)): by combining the two loops together we would have the situation where starting from one time parameter, with a non default units (eg `mjd`), an `astropyTime` object is attempted to be instantiated with the default format: this causes an exception.

Probably there is a different and better way to instantiate `astropyTime` objects, so I am open to suggestions.

Reply:
does it make sense?

Reply:
Not particularily, sorry. When I test actually find another issue, with `None` key being inserted in par_dic, and can not be jsonified. I am looking into it.

Reply:
Ok I see. It's because the true input unit is irreversibly overwritten when the first parameter of a given type is is processed. 

This is consequence of modifying arguments, which I think is often misleading (even if sometimes very beneficial).

Your approach is a way, I suggest to keep both input and output par_dic and substitute at the end. See #278

Reply:
Yes, that's the problem, it was not easy to explain. It comes out and it's clear when debugging.

What you suggest makes sense so we also avoid to modify the `par_dic`.